### ISSUE TITLE
Some question about vae results

### ISSUE THREAD

**Issue by Alienge** [Lightricks/LTX-Video#134]
Hello, thank you for your share elegant project!  I have some questions about the VAE encode-decode results.
I find that the VAE's reconstruction results after encoding and decoding are not  desired. Any advices?

**Result video**

https://github.com/user-attachments/assets/37ba661a-23f9-4a4f-9ff0-730a77c86789

**Code**
```
vframes, aframes, info = torchvision.io.read_video(filename="testvideo.mp4", pts_unit='sec', output_format='TCHW')
transform = transforms.Compose([
            video_transformer.ToTensorVideo(), # TCHW
            video_transformer.RandomHorizontalFlipVideo(),
            video_transformer.UCFCenterCropVideo((720,1280)),
            transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5], inplace=True)
        ])

frame_indice = np.linspace(0, 100, 9, dtype=int)
video = vframes[frame_indice]
x = transform(video)
x = torch.unsqueeze(x,dim=0).permute(0,2,1,3,4).cuda().half()
vae = AutoencoderKLLTXVideo.from_pretrained("Lightricks/LTX-Video", subfolder="vae", torch_dtype=torch.float16).to("cuda")
decode_noise_scale = torch.tensor([0.05]).cuda()
timestep = torch.tensor(decode_noise_scale, device="cuda", dtype=x.dtype)

decode_noise_scale = torch.tensor(decode_noise_scale, device="cuda", dtype=x.dtype)[:, None, None, None, None]
latent = vae.encode(x).latent_dist.sample()
noise = torch.randn(latent.shape,device="cuda", dtype=x.dtype,)

latents = (1 - decode_noise_scale) * latent + decode_noise_scale * noise


samples = vae.decode(latents, timestep, return_dict=False)[0]
samples = samples[0]
sample = ((samples * 0.5 + 0.5) * 255).add_(0.5).clamp_(0, 255).to(dtype=torch.uint8).cpu().permute(1, 2, 3, 0).contiguous()
index = 2

sample_save_path = f"./{index:04d}.mp4"
imageio.mimwrite(sample_save_path, sample, fps=8, quality=9)
```

**Comment by eisneim** [comment]
@Alienge why don't you follow what is doing in the inference.py?  init VAE like this: 

```
from ltx_video.models.autoencoders.causal_video_autoencoder import (
    CausalVideoAutoencoder,
)
from ltx_video.models.autoencoders.vae_encode import (
    get_vae_size_scale_factor,
    latent_to_pixel_coords,
    vae_decode,
    vae_encode,
)

vae = CausalVideoAutoencoder.from_pretrained("local_path/ltx-video-2b-v0.9.5.safetensors")

# you should use vae_decode, vae_encode

```

**Comment by Alienge** [comment]
@eisneim Thank you for your good advice. But I follow your advice , i am getting bad results. Do I norm it? But I see the source code in ltx_video.models.autoencoders.vae_encode and set the param "vae_per_channel_normalize =True " or "vae_per_channel_normalize =False" also get bad results. 
**Results**

https://github.com/user-attachments/assets/1c3a1c96-317d-4ca0-98c7-1811f67f39c1


**Code**
```
from ltx_video.models.autoencoders.vae_encode import (
    get_vae_size_scale_factor,
    latent_to_pixel_coords,
    vae_decode,
    vae_encode,
)

vframes, aframes, info = torchvision.io.read_video(filename="testvideo.mp4", pts_unit='sec', output_format='TCHW')

transform = transforms.Compose([
            video_transformer.ToTensorVideo(), # TCHW
            video_transformer.RandomHorizontalFlipVideo(),
            video_transformer.UCFCenterCropVideo((512,768)),
            transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5], inplace=True)
        ])

frame_indice = np.linspace(0, 100, 9, dtype=int)


video = vframes[frame_indice]

x = transform(video)

x = torch.unsqueeze(x,dim=0).permute(0,2,1,3,4).cuda().half()
vae = CausalVideoAutoencoder.from_pretrained("MODEL_DIR/ltx-video-2b-v0.9.5.safetensors").cuda().half()


latents = vae_encode(x,vae)

decode_noise_scale = 0.05
decode_noise_scale = torch.tensor([decode_noise_scale]).cuda()
decode_noise_scale = torch.tensor(decode_noise_scale, device="cuda", dtype=x.dtype)[:, None, None, None, None]
noise = torch.randn_like(latents)
latents = (latents * (1 - decode_noise_scale) + noise * decode_noise_scale)

x_hat = vae_decode(latents,vae,timestep=torch.tensor([0.05]).cuda())

samples = x_hat[0]
sample = ((samples * 0.5 + 0.5) * 255).add_(0.5).clamp_(0, 255).to(dtype=torch.uint8).cpu().permute(1, 2, 3, 0).contiguous()
index = 2

sample_save_path = f"./{index:04d}.mp4"
imageio.mimwrite(sample_save_path, sample, fps=8, quality=9)
```

**Comment by eisneim** [comment]
@Alienge you don't need to add noise to the latent to decode, i'v test it without adding noise, the result is good

**Comment by Alienge** [comment]
@eisneim I test with setting vae_per_channel_normalize to True. It doesnt work.  I think your metioned that "transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5], inplace=True)" is in the transform. 

**Results**

https://github.com/user-attachments/assets/cb209dde-cfdd-444e-b117-e516c54b3cc8


**code**
```
from ltx_video.models.autoencoders.vae_encode import (
    get_vae_size_scale_factor,
    latent_to_pixel_coords,
    vae_decode,
    vae_encode,
)

vframes, aframes, info = torchvision.io.read_video(filename="testvideo.mp4", pts_unit='sec', output_format='TCHW')

transform = transforms.Compose([
            video_transformer.ToTensorVideo(), # TCHW
            #video_transformer.RandomHorizontalFlipVideo(),
            video_transformer.UCFCenterCropVideo((512,768)),
            transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5], inplace=True)
        ])
frame_indice = np.linspace(0, 100, 9, dtype=int)
video = vframes[frame_indice]

x = transform(video)

x = torch.unsqueeze(x,dim=0).permute(0,2,1,3,4).cuda().half()
vae = CausalVideoAutoencoder.from_pretrained("MODEL_DIR/ltx-video-2b-v0.9.5.safetensors").cuda().half()
latents = vae_encode(x,vae,vae_per_channel_normalize=True)
x_hat = vae_decode(latents,vae,vae_per_channel_normalize=True,timestep=torch.tensor([0.05]).cuda())


samples = x_hat[0]
sample = ((samples * 0.5 + 0.5) * 255).add_(0.5).clamp_(0, 255).to(dtype=torch.uint8).cpu().permute(1, 2, 3, 0).contiguous()
index = 2
sample_save_path = f"./{index:04d}.mp4"
imageio.mimwrite(sample_save_path, sample, fps=8, quality=9)


```

**Comment by eisneim** [comment]
@Alienge your code might have issue, to verify it's not LTX vae's problem, try to swap it using COSMOS vae or CogvideoX vae or Hunyuan VAE

**Comment by Alienge** [comment]
@eisneim  If it is convenient, can you show me the code you tested? I haven't found the problem yet. When I load the weights of the ltx-video-2b-v0.9.safetensors version, the result is as shown in the figure below. But the result is not good.（same code from above. Just change below )
`vae = CausalVideoAutoencoder.from_pretrained("MODEL_DIR/ltx-video-2b-v0.9.safetensors").cuda().half()`


**Result**

https://github.com/user-attachments/assets/9217e6e9-89af-47c1-9039-6ba360add962

**Comment by Alienge** [comment]
@eisneim I got the result. It seems the loading parameters problem. I run the code by modifying the infer.py file in source code. And get results as following.   The result is not good. 
**Result**

https://github.com/user-attachments/assets/0737fd45-eba1-4ac0-b0ac-0224d06b834c

**Comment by eisneim** [comment]
@Alienge my code is here: [encode](https://github.com/eisneim/ltx_lora_training_i2v_t2v/blob/main/precompute.py)
and [decode](https://github.com/eisneim/ltx_lora_training_i2v_t2v/blob/main/test_rebuild_from_precomputed.py)
